### PR TITLE
[DPE-4121] Improvements on Spark E2E tests

### PR DESCRIPTION
**Pull request by theoctober19th**
This PR aims to:
1. Add a few integration tests related to Kyuubi bundle
2. Few minor improvements 
3. Add docstrings

### CODE REVIEW

**Review by deusebio — APPROVED** [review]
Great Job! LGTM!

File: python/tests/integration/conftest.py [hunk 1]
```diff
@@ -169,3 +183,50 @@ def create_file(path: Path, response: requests.Response):
 
         if cos_model:
             await ops_test.forget_model(cos_model)
+
+
+@pytest.fixture(scope="module")
+async def kyuubi_bundle(
```
Comment (welpaolo) [inline python/tests/integration/conftest.py:189]: I would like this to be renamed to spark_bundle

Comment (welpaolo) [reply python/tests/integration/conftest.py:189]: @deusebio what do you think ?

File: python/tests/integration/test_bundle.py [hunk 2]
```diff
@@ -54,103 +35,10 @@ def namespace(namespace_name):
...
-        apps=applications,
-        timeout=2500,
-        idle_period=30,
-        status="active",
-        raise_on_error=False,
-    )
-
-    for app in applications:
-        assert ops_test.model.applications[app].status == "active"
+async def test_deploy_bundle(ops_test, kyuubi_bundle):
```
Comment (welpaolo) [inline python/tests/integration/test_bundle.py:40]: This should be named `spark_bundle`

**Review by welpaolo — COMMENTED** [review]
All good, just want a clarification for @deusebio about how to name the bundle

File: python/tests/integration/conftest.py (hunk 1 above)
Comment (deusebio) [reply python/tests/integration/conftest.py:189]: Yes, that's a good idea!

Comment (theoctober19th) [reply python/tests/integration/conftest.py:189]: I agree. Just updated.

File: python/tests/integration/test_bundle.py (hunk 2 above)
Comment (theoctober19th) [reply python/tests/integration/test_bundle.py:40]: Update.

**Review by welpaolo — APPROVED** [review]
LGTM

**Pull request merged**